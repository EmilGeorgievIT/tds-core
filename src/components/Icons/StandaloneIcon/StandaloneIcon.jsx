import React from 'react'
import PropTypes from 'prop-types'

import safeRest from '../../../utils/safeRest'

import Icon from '../Icon/Icon'

import styles from './StandaloneIcon.modules.scss'

/**
 * An icon that has meaning within the context of the page, which should be communicated to screen readers.
 *
 * <span class="docs--badge__new">new!</span> <span class="docs--badge__version">v0.24.0</span>
 */
class StandaloneIcon extends React.Component {
  onClick = event => {
    const {onClick} = this.props

    if (onClick) {
      onClick(event)
    }
  }

  render() {
    const {symbol, variant, size, interactive, onClick, a11yText, ...rest} = this.props

    if (interactive) {
      return (
        <button {...safeRest(rest)} className={styles.interactive} aria-label={a11yText}>
          <Icon symbol={symbol} variant={variant} size={size} />
        </button>
      )
    }

    return <Icon {...rest} symbol={symbol} variant={variant} size={size} aria-label={a11yText} />
  }
}

StandaloneIcon.propTypes = {
  /**
   * Name of the icon symbol.
   */
  symbol: PropTypes.oneOf([
    'caretDown',
    'caretUp',
    'checkmark',
    'chevron',
    'leftChevron',
    'exclamationPointCircle',
    'expander',
    'hamburger',
    'location',
    'minus',
    'plus',
    'questionMarkCircle',
    'spyglass',
    'times',
  ]).isRequired,
  /**
   * The appearance of the Icon.
   */
  variant: PropTypes.oneOf(['primary', 'secondary', 'inverted', 'error']),
  /**
   * The icon size in pixels.
   */
  size: PropTypes.oneOf([16, 24, 48]),
  /**
   * Make the icon interactive
   */
  interactive: PropTypes.bool,
  /**
  * A callback function to be invoked when the interactive icon is clicked.
  */
  onClick: PropTypes.func,
  /**
   * A description of the icon for screen readers.
   */
  a11yText: PropTypes.string.isRequired,
}

StandaloneIcon.defaultProps = {
  variant: undefined,
  size: 24,
  interactive: false,
  onClick: undefined,
}

export default StandaloneIcon
