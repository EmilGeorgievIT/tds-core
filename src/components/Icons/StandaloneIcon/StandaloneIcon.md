### Available Icons

```
<div className="docs--horizontal-spacing">
  <StandaloneIcon symbol="caretDown" a11yText="text" />
  <StandaloneIcon symbol="caretUp" a11yText="text" />
  <StandaloneIcon symbol="checkmark" a11yText="text" />
  <StandaloneIcon symbol="leftChevron" a11yText="text" />
  <StandaloneIcon symbol="chevron" a11yText="text" />
  <StandaloneIcon symbol="exclamationPointCircle" a11yText="text" />
  <StandaloneIcon symbol="expander" a11yText="text" />
  <StandaloneIcon symbol="hamburger" a11yText="text" />
  <StandaloneIcon symbol="incomplete" a11yText="text" />
  <StandaloneIcon symbol="location" a11yText="text" />
  <StandaloneIcon symbol="minus" a11yText="text" />
  <StandaloneIcon symbol="plus" a11yText="text" />
  <StandaloneIcon symbol="questionMarkCircle" a11yText="text" />
  <StandaloneIcon symbol="spyglass" a11yText="text" />
  <StandaloneIcon symbol="times" a11yText="text" />
</div>
```
