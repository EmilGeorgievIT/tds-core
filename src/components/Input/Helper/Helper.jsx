import React from 'react'
import PropTypes from 'prop-types'

import safeRest from '../../../safeRest'
import ColoredTextProvider from '../../Typography/ColoredTextProvider/ColoredTextProvider'

import messagingStyles from '../../Messaging.modules.scss'
import styles from './Helper.modules.scss'

const getClassName = feedback => (feedback ? styles[feedback] : styles.default)

const renderContent = (feedback, children) => {
  if (feedback === 'error') {
    return (
      <ColoredTextProvider colorClassName={messagingStyles.errorText}>
        {children}
      </ColoredTextProvider>
    )
  }

  return children
}

const Helper = ({ feedback, children, ...rest }) => (
  <div {...safeRest(rest)} className={getClassName(feedback)}>
    {renderContent(feedback, children)}
  </div>
)
Helper.propTypes = {
  feedback: PropTypes.oneOf(['success', 'error']),
  children: PropTypes.node.isRequired
}
Helper.defaultProps = {
  feedback: undefined
}
Helper.displayName = 'Input.Helper'

export default Helper
