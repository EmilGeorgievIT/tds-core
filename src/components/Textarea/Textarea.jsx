import React from 'react'
import PropTypes from 'prop-types'
import { childrenOfType } from 'airbnb-prop-types'

import Tooltip from '../Tooltip/Tooltip'

import safeRest from '../../utils/safeRest'

import FormField from '../FormField/FormField'
import joinClassNames from '../../utils/joinClassNames'

class Textarea extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      value: props.value,
      focus: false,
    }
  }

  componentWillReceiveProps(nextProps) {
    if (this.state.value !== nextProps.value) {
      this.setState({
        value: nextProps.value,
      })
    }
  }

  onChange = event => {
    const { onChange } = this.props

    this.setState({
      value: event.target.value,
    })

    if (onChange) {
      onChange(event)
    }
  }

  onFocus = event => {
    const { onFocus } = this.props

    this.setState({ focus: true })

    if (onFocus) {
      onFocus(event)
    }
  }

  onBlur = event => {
    const { onBlur } = this.props

    this.setState({ focus: false })

    if (onBlur) {
      onBlur(event)
    }
  }

  render() {
    const { label, hint, feedback, error, helper, tooltip, ...rest } = this.props

    return (
      <FormField
        {...rest}
        value={this.state.value}
        label={label}
        hint={hint}
        focus={this.state.focus}
        feedback={feedback}
        error={error}
        helper={helper}
        tooltip={tooltip}
      >
        {(fieldId, ariaInvalid, ariaDescribedBy, className) => (
          // TODO: Remove the data-no-global-styles when getting rid of the global styles in _forms.scss
          <textarea
            {...safeRest(rest)}
            id={fieldId}
            className={joinClassNames(className)}
            value={this.state.value}
            onChange={this.onChange}
            onFocus={this.onFocus}
            onBlur={this.onBlur}
            aria-invalid={ariaInvalid}
            aria-describedby={ariaDescribedBy}
            data-no-global-styles
          >
            {this.state.value}
          </textarea>
        )}
      </FormField>
    )
  }
}

Textarea.propTypes = {
  /**
   * The label.
   */
  label: PropTypes.string.isRequired,
  /**
   * Clarify the expected input.
   */
  hint: PropTypes.string,
  /**
   * The value.
   */
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  /**
   * A feedback state.
   */
  feedback: PropTypes.oneOf(['success', 'error']),
  /**
   * An error message. Either an error or a helper should be used, not both.
   */
  error: PropTypes.string,
  /**
   * A detailed explanation of the input expected by a form field. Can be text,
   * other components, or HTML elements.
   *
   * If a function is provided, it must return a `Helper`. The function will be
   * invoked with the following arguments.
   *
   * @param {String} feedback The input's current feedback state.
   * @param {String} value The input's current value.
   */
  helper: PropTypes.oneOfType([PropTypes.func, PropTypes.node]),
  /**
   * A `Tooltip`
   */
  tooltip: childrenOfType(Tooltip),
  /**
   * A callback function to be invoked when the input value changes.
   *
   * @param {SyntheticEvent} event The React `SyntheticEvent`
   */
  onChange: PropTypes.func,
  /**
   * A callback function to be invoked when the input receives focus.
   *
   * @param {SyntheticEvent} event The React `SyntheticEvent`
   */
  onFocus: PropTypes.func,
  /**
   * A callback function to be invoked when the input loses focus.
   *
   * @param {SyntheticEvent} event The React `SyntheticEvent`
   */
  onBlur: PropTypes.func,
}

Textarea.defaultProps = {
  type: 'text',
  hint: undefined,
  value: '',
  feedback: undefined,
  error: undefined,
  helper: undefined,
  tooltip: undefined,
  onChange: undefined,
  onFocus: undefined,
  onBlur: undefined,
}

export default Textarea
