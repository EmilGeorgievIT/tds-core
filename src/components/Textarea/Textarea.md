### Minimal usage

When a form demands detailed content from the user, such as product feedback or support, then the Textarea component will fulfill that need.

```
<Textarea label="Enter some comments" onChange={e => console.log(e.target.value)}/>
```
