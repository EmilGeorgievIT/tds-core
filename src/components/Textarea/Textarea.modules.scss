.feedbackIconPosition {
  composes: absolute from '../Position.modules.scss';

  top: 0.5rem;
  right: 0.5rem;
}
