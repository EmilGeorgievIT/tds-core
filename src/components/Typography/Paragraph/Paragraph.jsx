import React from 'react'
import PropTypes from 'prop-types'

import safeRest from '../../../safeRest'

import spacingStyles from '../../Spacing.modules.scss'
import styles from './Paragraph.modules.scss'
import textStyles from '../Text/Text.modules.scss'

/**
 * <span class="docs--badge green">new!</span> <span class="docs--badge purple">v0.22.0</span>
 */
const Paragraph = ({ bold, size, align, invert, children, ...rest }, context) => {
  const paragraphColor = invert ? textStyles.invertedColor : textStyles.color

  const classes = `
    ${spacingStyles.noSpacing}
    ${context.inheritColor ? textStyles.inheritColor : paragraphColor}
    ${textStyles[size]}
    ${bold ? textStyles.boldFont : textStyles[`${size}Font`]}
    ${styles[`${align}Align`]}
    `

  return (
    <p {...safeRest(rest)} className={classes}>
      {children}
    </p>
  )
}

Paragraph.propTypes = {
  /**
   * Embolden paragraph text without conveying any special importance or relevance.
   */
  bold: PropTypes.bool,
  /**
   * Font size.
   */
  size: PropTypes.oneOf([
    'small',
    'medium',
    'large'
  ]),
  /**
   * Align content.
   */
  align: PropTypes.oneOf([
    'left',
    'center',
    'right'
  ]),
  /**
   * Invert to appear light on dark backgrounds.
   */
  invert: PropTypes.bool,
  /**
   * The content. Can be text, other components, or HTML elements.
   */
  children: PropTypes.node.isRequired
}

Paragraph.defaultProps = {
  bold: false,
  size: 'medium',
  align: 'left',
  invert: false
}

Paragraph.contextTypes = {
  inheritColor: PropTypes.bool
}

export default Paragraph
