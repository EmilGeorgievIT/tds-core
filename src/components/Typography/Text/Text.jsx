import React from 'react'
import PropTypes from 'prop-types'

import safeRest from '../../../utils/safeRest'
import TextSup from './TextSup/TextSup'
import TextSub from './TextSub/TextSub'

import styles from './Text.modules.scss'

/**
 * Produces inline text
 *
 * <span class="docs--badge green">new!</span> <span class="docs--badge purple">v0.22.0</span>
 */
const Text = ({ bold, size, invert, children, ...rest }, context) => {
  const textColor = invert ? styles.invertedColor : styles.color

  const classes = `
    ${styles[size]}
    ${bold ? styles.boldFont : styles[`${size}Font`]}
    ${context.inheritColor ? styles.inheritColor : textColor}
  `

  return (
    <span {...safeRest(rest)} className={classes}>
      {children}
    </span>
  )
}

Text.propTypes = {
  /**
   * Embolden text without conveying any special importance or relevance.
   */
  bold: PropTypes.bool,
  /**
   * Font size.
   */
  size: PropTypes.oneOf([
    'base',
    'small',
    'medium',
    'large'
  ]),
  /**
   * Invert span style to appear light on dark backgrounds.
   */
  invert: PropTypes.bool,
  /**
   * The text.
   */
  children: PropTypes.node.isRequired
}

Text.defaultProps = {
  bold: false,
  size: 'base',
  invert: false
}

Text.contextTypes = {
  inheritColor: PropTypes.bool
}

Text.Sup = TextSup
Text.Sub = TextSub

export default Text
