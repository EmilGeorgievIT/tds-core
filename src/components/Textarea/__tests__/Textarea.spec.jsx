import React from 'react'
import { render, mount } from 'enzyme'

import StandaloneIcon from '../../Icons/StandaloneIcon/StandaloneIcon'
import Text from '../../Typography/Text/Text'
import Paragraph from '../../Typography/Paragraph/Paragraph'
import Fade from '../../Animation/Fade'
import Textarea from '../Textarea'
import Helper from '../../Input/Helper/Helper'

describe('Textarea', () => {
  const defaultProps = {
    label: 'The textarea',
  }

  const doRender = (overrides = {}) => render(<Textarea {...defaultProps} {...overrides} />)

  const doMount = (overrides = {}) => {
    const textarea = mount(<Textarea {...defaultProps} {...overrides} />)

    const findTextareaElement = () => textarea.find('textarea')

    return {
      textarea,
      label: textarea.find('label'),
      findTextareaElement,
      findFeedbackIconFade: () => textarea.find(Fade),
      findHelper: () => textarea.find(Helper),
      changeValueTo: value => findTextareaElement().simulate('change', { target: { value } }),
      focus: (focusEvent = {}) => findTextareaElement().simulate('focus', focusEvent),
      blur: (blurEvent = {}) => findTextareaElement().simulate('blur', blurEvent),
    }
  }

  it('renders', () => {
    const textarea = doRender()

    expect(textarea).toMatchSnapshot()
  })

  it('renders with a feedback state and icon', () => {
    const textarea = doRender({ feedback: 'error' })

    expect(textarea).toMatchSnapshot()
  })

  describe('label', () => {
    it('must have a label', () => {
      const { label } = doMount({ label: 'The label' })

      expect(label).toContainReact(
        <Text size="medium" bold>
          The label
        </Text>
      )
    })

    it('can have a short hint', () => {
      const { label } = doMount({ hint: 'The short hint' })

      expect(label).toContainReact(<Text size="small">The short hint</Text>)
    })
  })

  describe('connecting the label to the textarea', () => {
    it('connects the label to the textarea', () => {
      const { label, findTextareaElement } = doMount()

      expect(label.prop('htmlFor')).toEqual(findTextareaElement().prop('id'))
    })

    it('uses the id when provided', () => {
      const { label, findTextareaElement } = doMount({
        id: 'the-id',
        name: 'the-name',
        label: 'The label',
      })

      expect(label).toHaveProp('htmlFor', 'the-id')
      expect(findTextareaElement()).toHaveProp('id', 'the-id')
    })

    it('uses the name when no id is provided', () => {
      const { label, findTextareaElement } = doMount({ name: 'the-name', label: 'The label' })

      expect(label).toHaveProp('htmlFor', 'the-name')
      expect(findTextareaElement()).toHaveProp('id', 'the-name')
    })

    it('generates an id from the label when no id or name is provided', () => {
      const { label, findTextareaElement } = doMount({ label: 'The label' })

      expect(label).toHaveProp('htmlFor', 'the-label')
      expect(findTextareaElement()).toHaveProp('id', 'the-label')
    })
  })

  describe('editability', () => {
    it('supports string values or number values', () => {
      let findTextareaElement = doMount().findTextareaElement
      expect(findTextareaElement()).toHaveValue('')

      findTextareaElement = doMount({ value: 'some value' }).findTextareaElement
      expect(findTextareaElement()).toHaveValue('some value')
    })

    it('has a value that can be changed', () => {
      const { changeValueTo, findTextareaElement } = doMount({ value: 'initial value' })
      changeValueTo('new value')

      expect(findTextareaElement()).toHaveValue('new value')
    })

    it('will notify when its value changes', () => {
      const onChangeMock = jest.fn()

      const { changeValueTo, findTextareaElement } = doMount({ onChange: onChangeMock })
      changeValueTo('new value')

      expect(onChangeMock).toHaveBeenCalledWith(
        expect.objectContaining({ target: { value: 'new value' } })
      )
      expect(findTextareaElement()).toHaveValue('new value')
    })

    it('can receive a new value from a parent component', () => {
      const { textarea, findTextareaElement } = doMount({ value: 'initial value' })

      textarea.setProps({ value: 'new value' })

      expect(findTextareaElement()).toHaveValue('new value')
    })
  })

  describe('focusing', () => {
    it('can be focused', () => {
      const { findTextareaElement, focus, blur } = doMount()
      expect(findTextareaElement()).toHaveClassName('default')

      focus()
      expect(findTextareaElement()).toHaveClassName('focus')

      blur()
      expect(findTextareaElement()).not.toHaveClassName('focus')
    })

    it('will notify when focus is gained', () => {
      const onFocusMock = jest.fn()
      const event = { target: { value: 'the value' } }

      const { focus } = doMount({ onFocus: onFocusMock })
      focus(event)

      // TODO: Need to do jest.matches or something to match only part of the event
      expect(onFocusMock).toHaveBeenCalledWith(expect.objectContaining(event))
    })

    it('will notify when focus is lost', () => {
      const onBlurMock = jest.fn()
      const event = { target: { value: 'the value' } }

      const { blur } = doMount({ onBlur: onBlurMock })
      blur(event)

      expect(onBlurMock).toHaveBeenCalledWith(expect.objectContaining(event))
    })
  })

  describe('feedback states', () => {
    it('can have a success feedback state', () => {
      const { findTextareaElement, findFeedbackIconFade } = doMount({ feedback: 'success' })

      expect(findTextareaElement()).toHaveClassName('success')
      expect(findFeedbackIconFade()).toContainReact(
        <StandaloneIcon
          symbol="checkmark"
          variant="primary"
          size={16}
          a11yText="The value of this input field is valid."
        />
      )
    })

    it('can have an error feedback state', () => {
      const { findTextareaElement, findFeedbackIconFade } = doMount({ feedback: 'error' })
      expect(findTextareaElement()).toHaveClassName('error')
      expect(findFeedbackIconFade()).toContainReact(
        <StandaloneIcon
          symbol="exclamationPointCircle"
          variant="error"
          size={16}
          a11yText="The value of this input field is invalid."
        />
      )
    })

    it('hides the feedback state while focused', () => {
      const { findTextareaElement, focus, blur } = doMount({ feedback: 'success' })

      focus()
      expect(findTextareaElement()).not.toHaveClassName('success')

      blur()
      expect(findTextareaElement()).toHaveClassName('success')
    })

    it('fades the feedback icon in on focus lost and out on focus gained', () => {
      const { findFeedbackIconFade, focus } = doMount({ feedback: 'success' })

      expect(findFeedbackIconFade()).toHaveProp('in', true)

      focus()
      expect(findFeedbackIconFade()).toHaveProp('in', false)
    })
  })

  it('can be disabled', () => {
    let findTextareaElement = doMount().findTextareaElement
    expect(findTextareaElement()).not.toHaveClassName('disabled')
    expect(findTextareaElement()).not.toBeDisabled()

    findTextareaElement = doMount({ disabled: true }).findTextareaElement
    expect(findTextareaElement()).toHaveProp('disabled')
    expect(findTextareaElement()).toBeDisabled()
  })

  it('can have an error message', () => {
    const { textarea } = doMount({ id: 'some-id', error: 'Oh no a terrible error!' })

    expect(textarea).toContainReact(
      <Helper id="some-id_error-message" feedback="error">
        <Paragraph size="small">Oh no a terrible error!</Paragraph>
      </Helper>
    )
  })

  describe('helpers', () => {
    it('can have a simple helper of some components', () => {
      const helper = <div>Some helper text.</div>
      const { textarea } = doMount({ id: 'some-id', helper })

      expect(textarea).toContainReact(
        <Helper id="some-id_helper">
          <Text size="small">
            <div>Some helper text.</div>
          </Text>
        </Helper>
      )
    })

    it('styles itself based on the textarea feedback state', () => {
      const helper = <Paragraph>Some helper text.</Paragraph>

      const { findHelper } = doMount({ feedback: 'success', helper })

      expect(findHelper()).toHaveProp('feedback', 'success')
    })

    it('can have a complex helper function to give control to the consumer', () => {
      const helper = jest.fn()
      helper.mockReturnValue(<Helper>Some helper text.</Helper>)

      const { textarea } = doMount({
        id: 'some-id',
        value: 'current value',
        feedback: 'error',
        helper,
      })

      expect(helper).toHaveBeenCalledWith('error', 'current value')
      expect(textarea).toContainReact(
        <div id="some-id_helper">
          <Text size="small">
            <Helper>Some helper text.</Helper>
          </Text>
        </div>
      )
    })
  })

  describe('accessibility', () => {
    it('marks the textarea as invalid when in the error feedback state', () => {
      let findTextareaElement = doMount().findTextareaElement
      expect(findTextareaElement()).toHaveProp('aria-invalid', false)

      findTextareaElement = doMount({ feedback: 'error' }).findTextareaElement
      expect(findTextareaElement()).toHaveProp('aria-invalid', true)
    })

    it('does not attach aria-describedby to the textarea field when no error or helper is present', () => {
      const { findTextareaElement } = doMount({ error: undefined, helper: undefined })

      expect(findTextareaElement()).toHaveProp('aria-describedby', undefined)
    })

    it('connects the error message to the textarea field for screen readers', () => {
      const { findTextareaElement, findHelper } = doMount({
        id: 'some-field-id',
        error: 'An error message',
      })

      expect(findTextareaElement()).toHaveProp('aria-describedby', 'some-field-id_error-message')
      expect(findHelper()).toHaveProp('id', 'some-field-id_error-message')
    })

    it('connects a simple helper to the textarea field for screen readers', () => {
      const helper = <Paragraph>Some helper text.</Paragraph>
      const { findTextareaElement, findHelper } = doMount({ id: 'some-field-id', helper })

      expect(findTextareaElement()).toHaveProp('aria-describedby', 'some-field-id_helper')
      expect(findHelper()).toHaveProp('id', 'some-field-id_helper')
    })

    it('connects a complex helper to the textarea field for screen readers', () => {
      const helper = () => <Helper>Complex helper</Helper>
      const { textarea, findTextareaElement } = doMount({ id: 'some-field-id', helper })

      expect(findTextareaElement()).toHaveProp('aria-describedby', 'some-field-id_helper')
      expect(textarea).toContainReact(
        <div id="some-field-id_helper">
          <Text size="small">
            <Helper>Complex helper</Helper>
          </Text>
        </div>
      )
    })
  })

  it('passes additional attributes to the textarea element', () => {
    const { findTextareaElement } = doMount({ name: 'a name', placeholder: 'a placeholder' })

    expect(findTextareaElement()).toHaveProp('name', 'a name')
    expect(findTextareaElement()).toHaveProp('placeholder', 'a placeholder')
  })

  it('does not allow custom CSS', () => {
    const { findTextareaElement } = doMount({
      className: 'my-custom-class',
      style: { color: 'hotpink' },
    })

    expect(findTextareaElement()).not.toHaveProp('className', 'my-custom-class')
    expect(findTextareaElement()).not.toHaveProp('style')
  })
})
