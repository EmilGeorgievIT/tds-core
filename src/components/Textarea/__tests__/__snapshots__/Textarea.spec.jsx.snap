// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea renders 1`] = `
<div
  class="betweenBottomMarginDesktop-2 stack"
>
  <div
    class="row justifyContentSpaceBetween relative"
  >
    <label
      data-no-global-styles="true"
      for="the-textarea"
    >
      <div
        class="betweenRightMarginDesktop-2 inline alignCenter"
      >
        <span
          class="medium boldFont color"
        >
          The textarea
        </span>
      </div>
    </label>
  </div>
  <div
    class="relative"
  >
    <textarea
      aria-invalid="false"
      class="default"
      data-no-global-styles="true"
      id="the-textarea"
    />
    <div
      class="feedbackIconPosition"
    />
  </div>
</div>
`;

exports[`Textarea renders with a feedback state and icon 1`] = `
<div
  class="betweenBottomMarginDesktop-2 stack"
>
  <div
    class="row justifyContentSpaceBetween relative"
  >
    <label
      data-no-global-styles="true"
      for="the-textarea"
    >
      <div
        class="betweenRightMarginDesktop-2 inline alignCenter"
      >
        <span
          class="medium boldFont color"
        >
          The textarea
        </span>
      </div>
    </label>
  </div>
  <div
    class="relative"
  >
    <textarea
      aria-invalid="true"
      class="error"
      data-no-global-styles="true"
      id="the-textarea"
    />
    <div
      class="feedbackIconPosition"
    >
      <div
        style="transition:opacity 100ms ease-in-out;opacity:1"
      >
        <div
          class="fixLineHeight"
        >
          <i
            aria-label="The value of this input field is invalid."
            class="iconExclamationPointCircle error size16"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
